### PR TITLE
Spoke store enforcer

### PR DESCRIPTION
**Pull request by Exide**
Does the following on startup unless ```spoke.enforceStore``` is false (defaults to true):

- Verifies ```spoke.path``` exists on the filesystem
- Looks for ```spoke.previousPaths``` in the config (comma delimited)
- Loops over the previous paths and, if they exist, merges them into the ```spoke.path``` location

### CODE REVIEW

**Comment by Exide** [comment]
Running more tests. Will reopen when complete.

**Comment by Exide** [comment]
I ran this against DLS with the following situations:

- Moved store to a child directory of the previous location
- Moved store to a parent directory of the previous location

Both were successful.

File: src/main/java/com/flightstats/hub/dao/file/FileUtil.java [hunk 1]
```diff
@@ -70,9 +72,57 @@ static boolean write(byte[] bytes, String filename, String path) {
...
+        List<Path> paths = Files.walk(source).collect(Collectors.toList());
+        for (int i = 0; i < paths.size(); ++i) {
+            Path path = paths.get(i);
+
+            if (!Files.exists(path)) {
+                logger.debug("skipping, already moved {}", path);
+                continue;
+            }
+
+            if (path.startsWith(destination)) {
```
Comment (moulliet) [inline src/main/java/com/flightstats/hub/dao/file/FileUtil.java]: I'm not sure this is correct.

Comment (Exide) [reply src/main/java/com/flightstats/hub/dao/file/FileUtil.java]: Agreed. Closing this PR again until I can make sure its bulletproof or not needed at all :p

**Pull request closed without merging**